src/typescript/12-generics/generics.md

```diff
@@ -370,61 +310,21 @@ root.right.value      //-> "Tom"
 root.right.left.value //-> "Matthew"
 ```
 
-### Verify Your Solution
-
-
-✏️ Run the following to verify your solution:
-
-```shell
-npm run 6-generics
-```
-
+<strong>Have issues with your local setup?</strong> You can use either [StackBlitz](https://stackblitz.com/fork/github/bitovi/academy/tree/main/exercises/typescript/12-generics/02-problem?file=src/tree-node.ts) or [CodeSandbox](https://codesandbox.io/p/devbox/github/bitovi/academy/tree/main/exercises/typescript/12-generics/02-problem?file=src/tree-node.ts) to do this exercise in an online code editor.
 
-### The solution
+### Solution 2
 
 <details>
 <summary>Click to see the solution</summary>
 
-✏️ Update `6-tree-node.ts` to the following:
+Update `tree-node.ts` to the following:
 
-```typescript
-interface Comparison<T> {
-    (v1:T,v2: T): number;
-}
+@sourceref ../../../exercises/typescript/12-generics/02-solution/src/tree-node.ts
+@highlight 5-9, 11, 16, only
 
-class TreeNode<T> {
-
-	value: T;
-	compare: Comparison<T>;
-	left?: TreeNode<T>;
-	right?: TreeNode<T>;
-
-	constructor(val: T, compare: Comparison<T> ) {
-		this.value = val;
-		this.compare = compare;
-	}
-
-	add(val: T){
-		if( this.compare(this.value, val) >= 1 ) {
-			if(this.left == null) {
-				this.left = new TreeNode(val, this.compare);
-			} else {
-				this.left.add(val);
-			}
-
-		} else {
-			if(this.right == null) {
-				this.right = new TreeNode(val, this.compare);
-			} else {
-				this.right.add(val);
-			}
-		}
-	}
-}
-
-export default TreeNode;
-```
+As we use generics in line 5, to utilize it in the rest of the `TreeNode` Class
```

Comment:
I’m not 100% sure what this is saying, can it be made more clear?

Reply:
LIke this? "As we use generics in line 5, we allow the `TreeNode` class to be flexible and reusable, accommodating different types of data and comparison logic."